paddle/fluid/operators/cast_op.cc

```diff
@@ -77,6 +81,27 @@ class CastOp : public framework::OperatorWithKernel {
     if (platform::is_cuda_pinned_place(tensor_place)) {
       return framework::OpKernelType(tensor->type(), ctx.device_context());
     }
+
+#ifdef PADDLE_WITH_MKLDNN
+    int in_dtype = ctx.Attr<int>("in_dtype");
+    int out_dtype = ctx.Attr<int>("out_dtype");
+
+    auto MKLDNNSupportsCast = [&]() -> bool {
+      int dtype_fp32 = (int)framework::proto::VarType::FP32;
+      int dtype_bf16 = (int)framework::proto::VarType::BF16;
+
+      if (in_dtype != dtype_fp32 && in_dtype != dtype_bf16) return false;
```

Comment:
Maybe you could use **or** logic between both ifs to have unified boolean condition.

Reply:
Nice idea, I'll do that